samza-api/src/main/java/org/apache/samza/checkpoint/CheckpointId.java

```diff
@@ -61,6 +61,11 @@ public long getNanos() {
     return nanos;
   }
 
+  /**
+   * WARNING: Do not change the toString() representation. It is used for serde'ing {@link CheckpointId} as part of task
+   * checkpoints, in conjunction with {@link #fromString(String)}.
+   * @return the String representation of this {@link CheckpointId}.
+   */
```

Comment:
@mynameborat It didn't seem like it was worth adding the overhead of a separate serde class with the associated ceremony (serde factory, classload serde depending on the class version, etc.) for a relatively simple operation. The interface also isn't exactly the same (since this serdes to/from string while the Serde interfaces use byte[]). 

It makes sense to rename them to 'to/from' or 'serialize/deserialize'.

Reply:
@dxichen 
> use the jackson model for here and all other occurences
Is that backwards compatible?

Reply:
We would have to accept both versions in the fromString/deserialize method to perserve backwards compatibility